[BUG] Snapshot build is broken

**Describe the bug**

Snapshot build is broken.

**To Reproduce**

```
./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=1.0.0-SNAPSHOT -Dcommon_utils.version=1.0.0.0-SNAPSHOT -Djob_scheduler.version=1.0.0.0-SNAPSHOT -Dbuild.snapshot=true
```

Fails as follows.

```
> Evaluating settings
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/ubuntu/.gradle/wrapper/dists/gradle-6.6.1-all/ejrtlte9hlw8v6ii20a9584rs/gradle-6.6.1/
lib/groovy-all-1.3-2.5.12.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
=======================================
OpenSearch Build Hamster says Hello!
  Gradle Version        : 6.6.1
  OS Info               : Linux 5.8.0-1041-aws (amd64)
  JDK Version           : 14 (JDK)
  JAVA_HOME             : /usr/lib/jvm/java-14-openjdk-amd64
  Random Testing Seed   : 10B85CB25E74D133
  In FIPS 140 mode      : false
=======================================
> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find org.opensearch:opensearch-job-scheduler-spi:1.0.0.0.
     Searched in the following locations:
       - file:/home/ubuntu/.m2/repository/org/opensearch/opensearch-job-scheduler-spi/1.0.0.0/opensearch-job-scheduler-spi-1.0.0.0.pom
       - https://repo.maven.apache.org/maven2/org/opensearch/opensearch-job-scheduler-spi/1.0.0.0/opensearch-job-scheduler-spi-1.0.0.0.pom
       - https://plugins.gradle.org/m2/org/opensearch/opensearch-job-scheduler-spi/1.0.0.0/opensearch-job-scheduler-spi-1.0.0.0.pom
       - https://jcenter.bintray.com/org/opensearch/opensearch-job-scheduler-spi/1.0.0.0/opensearch-job-scheduler-spi-1.0.0.0.pom
     Required by:
         project :
```

**Expected behavior**

Work, produce `anomaly-detection-1.0.0.0-SNAPSHOT.zip`.

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -33,7 +33,7 @@ jobs:
         with:
           repository: 'opensearch-project/OpenSearch'
           path: OpenSearch
-          ref: 'main'
+          ref: '1.x'
       - name: Build OpenSearch
         working-directory: ./OpenSearch
         run: ./gradlew publishToMavenLocal
@@ -59,11 +59,11 @@ jobs:
 
       - name: Build job-scheduler
         working-directory: ./job-scheduler
-        run: ./gradlew publishToMavenLocal -Dopensearch.version=1.1.0-SNAPSHOT -Dbuild.snapshot=false
+        run: ./gradlew publishToMavenLocal -Dopensearch.version=1.1.0-SNAPSHOT
       - name: Assemble job-scheduler
         working-directory: ./job-scheduler
         run: |
-          ./gradlew assemble -Dopensearch.version=1.1.0-SNAPSHOT -Dbuild.snapshot=false
+          ./gradlew assemble -Dopensearch.version=1.1.0-SNAPSHOT
           echo "Creating ../src/test/resources/job-scheduler ..."
           mkdir -p ../src/test/resources/job-scheduler
           pwd
@@ -83,17 +83,14 @@ jobs:
 
       - name: Multi Nodes Integration Testing
         run: |
-          ./gradlew integTest  -PnumNodes=3
+          ./gradlew integTest -PnumNodes=3
 
       - name: Pull and Run Docker
         run: |
-          ## plugin=`ls build/distributions/*.zip`
-          ## version=`echo $plugin|awk -F- '{print $4}'| cut -d. -f 1-3`
-          ## plugin_version=`echo $plugin|awk -F- '{print $4}'| cut -d. -f 1-4`
-          ## TODO: remove these two hard code versions below after GA release
+          plugin=`ls build/distributions/*.zip`
           version=1.1.0-SNAPSHOT
-          plugin_version=1.1.0.0
-          echo $version
+          plugin_version=1.1.0.0-SNAPSHOT
+          echo Using OpenSearch $version with AD $plugin_version
           cd ..
           if docker pull opensearchstaging/opensearch:$version
           then

--- a/build.gradle
+++ b/build.gradle
@@ -31,8 +31,8 @@ import org.opensearch.gradle.testclusters.StandaloneRestIntegTestTask
 buildscript {
     ext {
         opensearch_group = "org.opensearch"
-        opensearch_version = System.getProperty("opensearch.version", "1.0.0")
-        // 1.0.0 -> 1.0.0.0, and 1.0.0-SNAPSHOT -> 1.0.0.0-SNAPSHOT
+        opensearch_version = System.getProperty("opensearch.version", "1.1.0-SNAPSHOT")
+        // 1.1.0 -> 1.1.0.0, and 1.1.0-SNAPSHOT -> 1.1.0.0-SNAPSHOT
         opensearch_build = opensearch_version.replaceAll(/(\.\d)([^\d]*)$/, '$1.0$2')
         common_utils_version = System.getProperty("common_utils.version", opensearch_build)
         job_scheduler_version = System.getProperty("job_scheduler.version", opensearch_build)
@@ -79,7 +79,7 @@ ext {
 allprojects {
     group = 'org.opensearch'
 
-    version = "${opensearch_version}" - "-SNAPSHOT" + ".0"
+    version = opensearch_version - "-SNAPSHOT" + ".0"
     if (isSnapshot) {
         version += "-SNAPSHOT"
     }

--- a/build.gradle
+++ b/build.gradle
@@ -31,9 +31,11 @@ import org.opensearch.gradle.testclusters.StandaloneRestIntegTestTask
 buildscript {
     ext {
         opensearch_group = "org.opensearch"
-        opensearch_version = System.getProperty("opensearch.version", "1.1.0-SNAPSHOT")
-        common_utils_version = System.getProperty("common_utils.version", "1.1.0.0")
-        job_scheduler_version = System.getProperty("job_scheduler.version", "1.1.0.0")
+        opensearch_version = System.getProperty("opensearch.version", "1.0.0")
+        // 1.0.0 -> 1.0.0.0, and 1.0.0-SNAPSHOT -> 1.0.0.0-SNAPSHOT
+        opensearch_build = opensearch_version.replaceAll(/(\.\d)([^\d]*)$/, '$1.0$2')
+        common_utils_version = System.getProperty("common_utils.version", opensearch_build)
+        job_scheduler_version = System.getProperty("job_scheduler.version", opensearch_build)
     }
 
     repositories {
@@ -62,11 +64,6 @@ repositories {
     jcenter()
 }
 
-ext {
-    opensearchVersion = System.getProperty("opensearch.version", "1.1.0")
-    isSnapshot = "true" == System.getProperty("build.snapshot", "true")
-}
-
 apply plugin: 'java'
 apply plugin: 'idea'
 apply plugin: 'opensearch.opensearchplugin'
@@ -74,6 +71,10 @@ apply plugin: 'opensearch.testclusters'
 apply plugin: 'base'
 apply plugin: 'jacoco'
 apply plugin: 'eclipse'
+
+ext {
+    isSnapshot = "true" == System.getProperty("build.snapshot", "true")
+}
 
 allprojects {
     group = 'org.opensearch'

--- a/build.gradle
+++ b/build.gradle
@@ -67,8 +67,6 @@ ext {
     isSnapshot = "true" == System.getProperty("build.snapshot", "true")
 }
 
-version = "${opensearchVersion}.0"
-
 apply plugin: 'java'
 apply plugin: 'idea'
 apply plugin: 'opensearch.opensearchplugin'
@@ -79,6 +77,11 @@ apply plugin: 'eclipse'
 
 allprojects {
     group = 'org.opensearch'
+
+    version = "${opensearch_version}" - "-SNAPSHOT" + ".0"
+    if (isSnapshot) {
+        version += "-SNAPSHOT"
+    }
 
     plugins.withId('java') {
         sourceCompatibility = targetCompatibility = "1.8"

